### PR TITLE
[3006.x] Migrate workflows off AWS

### PR DESCRIPTION
**Pull request by dwoz**
### What does this PR do?

* Migrate off aws runners
   * Use github enterprise runners for linux based workflows in place of self hosted runners on aws
* Migrate off S3 caching for workflows
    * Use github's cache instead of s3
    *  We may need to cut back on what we cache. iirc, github's cache was not scaling for us.

### What issues does this PR fix or reference?
Fixes

### Previous Behavior
Remove this section if not relevant

### New Behavior
Remove this section if not relevant

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
Yes/No

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices, including the
[PR Guidelines](https://docs.saltproject.io/en/master/topics/development/pull_requests.html).

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Review by dmurphy18 — DISMISSED** [review]
Need to clean up pre-commit
Wondering about distinct lack of Arm64, is everything going to be x86_64 ?

**Comment by dwoz** [comment]
> Need to clean up pre-commit Wondering about distinct lack of Arm64, is everything going to be x86_64 ?

No, we have `linux-x86_64` and `linux-arm64` runners. I just needed to add them to the linter config.

**Review by twangboy — DISMISSED** [review]

File: .github/workflows/test-action-windows.yml [hunk 1]
```diff
@@ -88,7 +86,7 @@ jobs:
     steps:
       - uses: actions/setup-python@v5
         with:
-          python-version: '3.10'
+          python-version: 3.10.14
```
Comment (twangboy) [inline .github/workflows/test-action-windows.yml:89]: We created a new variable "gh-actions-python-version" above... would we want to use that here?

File: .github/workflows/test-action-windows.yml [hunk 2]
```diff
@@ -127,9 +122,11 @@ jobs:
         include: ${{ fromJSON(needs.generate-matrix.outputs.matrix-include) }}
     env:
       SALT_TRANSPORT: ${{ matrix.transport }}
-      TEST_GROUP: ${{ matrix.test-group || 1 }}
 
     steps:
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.10'
```
Comment (twangboy) [inline .github/workflows/test-action-windows.yml:129]: Probably here as well

File: .github/workflows/test-packages-action-windows.yml [hunk 3]
```diff
@@ -78,9 +78,7 @@ jobs:
     runs-on:
       # We need to run on our self-hosted runners because we need proper credentials
       # for boto3 to scan through our repositories.
-      - self-hosted
-      - linux
-      - x86_64
+      - ubuntu-latest
```
Comment (twangboy) [inline .github/workflows/test-packages-action-windows.yml:81]: Aren't we running directly on Windows now?

Comment (dwoz) [reply .github/workflows/test-packages-action-windows.yml:81]: This is for the test martrix step.

File: .github/workflows/test-action-windows.yml [hunk 4]
```diff
@@ -1,5 +1,5 @@
 ---
-name: Test Artifact
+name: Test Artifact(macOS)
```
Comment (twangboy) [inline .github/workflows/test-action-windows.yml:2]: Windows

**Review by twangboy — CHANGES_REQUESTED** [review]

File: .github/workflows/build-packages.yml [hunk 5]
```diff
@@ -340,19 +218,128 @@ jobs:
...
+            MSG="The sign-macos-packages input is false. The packages created will NOT be signed."
+            echo "${MSG}"
+            echo "${MSG}" >> "${GITHUB_STEP_SUMMARY}"
+            echo "sign-pkgs=false" >> "$GITHUB_OUTPUT"
+          fi
+
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: 3.11
```
Comment (dmurphy18) [inline .github/workflows/build-packages.yml:260]: wondering about py 3.11 and if it needs quotes for safety, understanding not a 3.10 aka 3.1zero issue but defensive programming

**Review by dmurphy18 — APPROVED** [review]
Wonder about using  Python 3.11 on MacOS
And probably leaving python-version: "3.10.15", to 3.10.16 till after everything working

**Pull request merged**